app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,24 @@
+class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  def twitter
+    @user = User.from_omniauth_twitter(request.env["omniauth.auth"])
+    if @user.persisted?
+      sign_in_and_redirect @user, :event => :authentication #this will throw if @user is not activated
+      set_flash_message(:notice, :success, :kind => "Twitter") if is_navigational_format?
+    else
+      session["devise.twitter_data"] = request.env["omniauth.auth"]
+      redirect_to new_user_registration_url
+    end
+  end
+
+ def facebook
```

Comment:
Assignment Branch Condition size for facebook is too high. [15.17/15]<br>Inconsistent indentation detected.